src/robots/msj_platform.cpp

```diff
@@ -157,11 +159,52 @@ class MsjPlatform: public cardsflow::kindyn::Robot{
 
     }
 
+    ///find the closest limit point when the robot is in infeasible state
+    VectorXd findClosestJointLimit(double q0, double q1, double q3){
+        ROS_INFO("FINDING CLOSEST JOINT LIMIT");
+        VectorXd closestLimit;
+        closestLimit.resize(number_of_dofs);
+        closestLimit.setZero();
+        double distance = 0.0, smallestDistance = numeric_limits<double>::max();
+
+        for(int i=0; i < limits[0].size(); i++){
+            VectorXd jointAngle = Vector2d::Zero(), jointLimits = Vector2d::Zero();
+            jointAngle << q0, q1;
+            jointLimits << limits[0][i] ,limits[1][i];
+            distance = (jointAngle - jointLimits).norm();
+            if (distance < smallestDistance){
+                smallestDistance = distance;
+                closestLimit[0] = jointLimits[0];
+                closestLimit[1] = jointLimits[1];
+                closestLimit[2] = q[2];
```

Comment:
Should we be using here q3 instead of accessing q[2] ??

Reply:
true